src/core/server/saved_objects/so_types/config.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectsType } from '../types';
+
+export const config: SavedObjectsType = {
+  name: 'config',
+  hidden: false,
+  namespaceAgnostic: false,
+  mappings: {
+    dynamic: true as any, // TODO: check if can be switched to false, else adapt the type to allow true
```

Comment:
I'm not sure if we use it, but with dynamic: false we wouldn't be able to do telemetry on configuration objects which might be useful information.

Reply:
I'm too afraid to touch that value anyway, this will stay `true`.

Now the question is: do we want to allow plugins to register types with `dynamic=true` mappings? As josh suggested, we could have our own internal type representation that allow `true` where it's not allowed in the public `SavedObjectType`, however this means a lot of changes for something that can be force-casted anyway if plugins really wants to do it, so I'm leaning to just change `false | 'strict'` to `boolean | strict`. @joshdover wdyt?

Reply:
> I'm not sure if we use it, but with dynamic: false we wouldn't be able to do telemetry on configuration objects which might be useful information.

I don't really think this has to be implemented as a ES query even if we did have telemetry like this. We only have one of these objects per space, per release version. I'm not sure how many spaces we support, but I can't imagine a customer ever having so many that doing this calculation in-memory would be a problem.

That said, it may not be worth doing this change right now.

> so I'm leaning to just change `false | 'strict'` to `boolean | strict`

I lean towards just keeping it the way you have it (force casted) so as to not encourage plugins to use `dynamic: true` since it is probably never what we want or need. If a legitimate use case comes up, we can change this.

Reply:
Ok, changing back to force-casting with explanatory comment